frontend/src/pages/AccountReceive.vue

```diff
@@ -60,21 +60,21 @@
         </q-card>
```

Comment:
Hey @garyghayrat, this seems like a great start. It's definitely a better experience to only have to wait for a small, recent tranche of announcements to find recent transactions. There are some UX issues we're going to have to figure out though.

First, I'm not a fan of the way the second tranche of scanning takes place with a full UI below the table. It's a bit confusing. It's also not accurate to show the message that says "Scanning Completed Successfully" while the scanning is still ongoing.

There are lots of ways we could solve this, but here's one mockup I through together:

<img width="1232" alt="scanning-mockup" src="https://github.com/ScopeLift/umbra-protocol/assets/1479802/e7a6c0fd-340a-4285-a9fe-e183e79061f0">

Another issue is what happens when the second tranche of scanning completes. As it is now, the full table completely reloads. This leads to really bad UX of the screen flashing, and it's very problematic if the user is in the middle of attempting or completing a withdraw, because the forms/modals are lost. See this video to see what I mean:

![receive-reset-issues](https://github.com/ScopeLift/umbra-protocol/assets/1479802/7a1754b7-d5a2-42da-b5d0-eafde6ef5571)

I think the right way to do this is to *update* the table contents with additional records after they're found rather than completely reseting/re-rendering the table. I'm not sure if this is easy to do with the components we're using, but does it make sense?

Reply:
That makes complete sense, and love the mockup. 
> update the table contents with additional records after they're found rather than completely reseting/re-rendering the table. 
I'll look into this more and iterate, then I'll get back to you!

Reply:
Given that web workers run in parallel, and we want to update the table in place to prevent jumping around or other UX issues, does this result in us losing ordering guarantees in the table? Web worker 2 might return logs 1000-2000 before web worker 1 finishes returning logs 0-1000, so I think we have to choose between "stable table ordering when events come in" and "sort table by timestamp"?

Reply:
I think to giving more context before answering those questions will be helpful here. 
So the initial ideal goal was to scan every 1000 announcements as they come in and whenever new announcements are found, update the table with the new user announcements.

Challenges there were that creating and teriminating web workers cost resources and time, and what was actually happening was, before the first batch of maximum allowed web workers are created and have the time to process the first 1000 announcements, 10-20 requests to create new sets of workers all trying to process their share of 1000 announcements were taking so much resource and time that it actually slowed down the whole process. 

I tried using a library like [workerpool](https://www.npmjs.com/package/workerpool) which was suppose to help create a pool of workers, and instead of terminating them once they done scanning (terminating and creating workers take resources), they would wait for new announcements to be added to their task queue. But after installing that library, I'm assuming because we are trying to call pieces of code from a dependency like `umbra-js` I had webpack related issues that I thought wasn't worth the time investigating. 

Ben suggested splitting up the process in 2 instead, where you first scan 10,000 announcements, which should cover a reasonable timeframe where most of our users will have received funds, and then scan rest of the announcements.

Reply:
Moving this convo to https://github.com/ScopeLift/umbra-protocol/pull/558#discussion_r1267233356 to avoid derailing this thread